SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Utils/Interest/ResultTypes/ResultTypesTest.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+#include "Mocks/SpatialClassInfoManagerMock.h"
+
+#include "Utils/Interest/ResultTypes.h"
+
+#define RESULT_TYPES_TEST(TestName) \
+	GDK_TEST(Core, ResultTypes, TestName)
+
+// Sanity check tests for creating result types.
+// Run tests inside the SpatialGDK namespace in order to test static functions.
+namespace SpatialGDK
+{
+	TArray<Worker_ComponentId> DataComponentIds = { 1, 2 };
+	TArray<Worker_ComponentId> OwnerOnlyComponentIds = { 3, 4 };
+	TArray<Worker_ComponentId> HandoverComponentIds = { 5, 6 };
+
+	USpatialClassInfoManagerMock* CreateAndPopulateMockClassInfoManager()
+	{
+		USpatialClassInfoManagerMock* ClassInfoManager = NewObject<USpatialClassInfoManagerMock>();
+
+		// Ensure the mock won't be garbage collected
+		ClassInfoManager->AddToRoot();
```

Comment:
Does this need to be removed at the end of each test? If so I'd recommend making this a unique_ptr with a deleter that does that.

Reply:
turns out using smart pointers on Uobjects leads to all kinds of issues. AFAIU of unreal's memory management, I will achieve what I want by simply removing this line, as the garbage collector should handle the cleanup